### PR TITLE
added mzQC export for the QCCalculator

### PR DESCRIPTION
**Pull request by axelwalter**
# Description

Added a function for QCFile to export as mzQC in JSON format. 

# TODO:
- calculate SHA
- add option to store as qcML or mzQC
- add tests / test files

### CODE REVIEW

File: src/openms/source/FORMAT/QcMLFile.cpp [hunk 1]
```diff
@@ -1202,6 +1272,150 @@ namespace OpenMS
...
+        << String(14, ' ') << "\"fileProperties\": [\n"
+        << String(16, ' ') << "{\n"
+        << String(18, ' ') << "\"accession\": \"MS:1000747\",\n" // exclude in qualityMetrics?
+        << String(18, ' ') << "\"name\": \"completion time\",\n"
+        << String(18, ' ') << "\"value\": \"" + completionTime + "\"\n"
+        << String(16, ' ') << "},\n"
+        << String(16, ' ') << "{\n"
+        << String(18, ' ') << "\"accession\": \"MS:1000569\",\n"
+        << String(18, ' ') << "\"name\": \"SHA-1\",\n"
+        << String(18, ' ') << "\"value\": \"" << "\"\n" // how to calculate SHA sum?
```
Comment (timosachsenberg) [inline src/openms/source/FORMAT/QcMLFile.cpp:1376]: from MSExperiment you can get the SourceFile https://github.com/OpenMS/OpenMS/blob/develop/src/tests/class_tests/openms/source/SourceFile_test.cpp#L150

**Review by timosachsenberg — CHANGES_REQUESTED** [review]
Nice thanks for the contribution

**Comment by cbielow** [comment]
interesting PR.
Can you motivate a bit why mzQC is added into the old qcML architecture?

The way I see it (and I think @mwalzer agrees): qcML is dead, so my first take would have been to start mzQC from scratch instead of trying to adapt old code. But that's just me.

We also have https://github.com/nlohmann/json already integrated into OpenMS, to immediate use.
To write a JSON string, you can just do (taken from the example section of `json`):

```
json j2 = {
  {"pi", 3.141},
  {"happy", true},
  {"name", "Niels"},
  {"nothing", nullptr},
  {"answer", {
    {"everything", 42}
  }},
  {"list", {1, 0, 2}},
  {"object", {
    {"currency", "USD"},
    {"value", 42.99}
  }}
};
```
This would avoid the need for manually stitching everything together and gives the correct indentations automatically etc...

The final remark: the official file ending of mzQC will be `.mzQC` (I think this is not part of the spec document yet, but we agreed upon that in the last mzQC meeting, if I remember correctly).

**Comment by cbielow** [comment]
I'm happy to discuss more in a TC. maybe in the afternoon?

**Comment by axelwalter** [comment]
Thanks for the feedback, we can discuss it in the afternoon meeting.

**Comment by axelwalter** [comment]
Closing this pull request, revert changes and create new MzQCFile and MzQCCalculator independently of the old qcML stuff.

**Pull request closed without merging**